pkg/mimirpb/compat_test.go

```diff
@@ -223,16 +222,50 @@ func TestFromFPointsToSamples(t *testing.T) {
 	assert.Equal(t, expected, FromFPointsToSamples(input))
 }
 
+// Check that Prometheus FPoint and Mimir Sample types converted
+// into each other with unsafe.Pointer are compatible
+func TestPrometheusFPointInSyncWithMimirPbSample(t *testing.T) {
+	test.RequireSameShape(t, promql.FPoint{}, Sample{}, true)
+}
+
+func BenchmarkFromFPointsToSamples(b *testing.B) {
+	n := 100
+	input := make([]promql.FPoint, n)
+	for i := 0; i < n; i++ {
+		input[i] = promql.FPoint{T: int64(i), F: float64(i)}
+	}
+	for i := 0; i < b.N; i++ {
+		FromFPointsToSamples(input)
+	}
+}
+
 func TestFromHPointsToHistograms(t *testing.T) {
 	input := []promql.HPoint{{T: 3, H: test.GenerateTestFloatHistogram(0)}, {T: 5, H: test.GenerateTestFloatHistogram(1)}}
 	expected := []FloatHistogramPair{
-		{TimestampMs: 3, Histogram: *FloatHistogramFromPrometheusModel(test.GenerateTestFloatHistogram(0))},
-		{TimestampMs: 5, Histogram: *FloatHistogramFromPrometheusModel(test.GenerateTestFloatHistogram(1))},
+		{TimestampMs: 3, Histogram: FloatHistogramFromPrometheusModel(test.GenerateTestFloatHistogram(0))},
+		{TimestampMs: 5, Histogram: FloatHistogramFromPrometheusModel(test.GenerateTestFloatHistogram(1))},
 	}
 
 	assert.Equal(t, expected, FromHPointsToHistograms(input))
 }
 
+// Check that Prometheus HPoint and Mimir FloatHistogramPair types converted
+// into each other with unsafe.Pointer are compatible
+func TestPrometheusHPointInSyncWithMimirPbFloatHistogramPair(t *testing.T) {
+	test.RequireSameShape(t, promql.HPoint{}, FloatHistogramPair{}, true)
+}
+
+func BenchmarkFromHPointsToHistograms(b *testing.B) {
+	n := 100
+	input := make([]promql.HPoint, n)
+	for i := 0; i < n; i++ {
+		input[i] = promql.HPoint{T: int64(i), H: test.GenerateTestFloatHistogram(i)}
+	}
+	for i := 0; i < b.N; i++ {
```

Comment:
Same as above: we should call `b.ResetTimer()` before entering the benchmark loop.